### PR TITLE
Add a ScalingSchedule metrics for autoscaler

### PR DESCRIPTION
**Pull request by jonathanbeber**
This commit creates integration with the [`ScalingSchedule`
collectors][0] for autoscaler adding two new metric types:
`ScalingSchedule` and `ClusterScalingSchedule`. The metrics
configuration are pretty simple and receives just the object name.

[0]: https://github.com/zalando-incubator/kube-metrics-adapter#scalingschedule-collectors

### CODE REVIEW

File: pkg/core/autoscaler.go [hunk 1]
```diff
@@ -292,6 +302,62 @@ func zmonMetric(metrics zv1.AutoscalerMetrics, stackName, namespace string) (*au
...
+		return nil, fmt.Errorf("average not specified")
+	}
+	average := metrics.Average.DeepCopy()
+
+	if metrics.ScalingSchedule == nil {
+		return nil, errMissingScalingScheduleDefinition
+	}
+
+	name := metrics.ScalingSchedule.Name
+	if name == "" {
```
Comment (mikkeloscar) [inline pkg/core/autoscaler.go:316]: This we could also catch already in the CRD validation with a pattern similar to what is used for validating `metadata.name`

Comment (jonathanbeber) [reply pkg/core/autoscaler.go:316]: We do, right? I didn't set this field as optional, so I assume it's done, but I find a weak link to do not validate here. If you think we could just drop it?

Comment (jonathanbeber) [reply pkg/core/autoscaler.go:316]: hmm, I got it now, it indeed allows me to set an empty string. I will add the validation. Apart from that, do you think we should/could drop this validation here?

**Comment by jonathanbeber** [comment]
@mikkeloscar thanks for taking a look, I added a CRD pattern validation, the same used by the `metadata.name` as you suggested.

File: pkg/core/autoscaler.go (hunk 1 above)
Comment (mikkeloscar) [reply pkg/core/autoscaler.go:316]: > so both might be fine.

Yes, both are fine, but it's a better user experience to catch it sooner.

**Comment by mikkeloscar** [comment]
:+1:

**Comment by jonathanbeber** [comment]
:+1:

**Pull request merged**